core/api/transport/session.hpp

```diff
@@ -106,6 +106,13 @@ namespace kagome::api {
      */
     virtual SessionType type() const = 0;
 
+    /**
+     * `boost::asio::post(io_context, cb)` to `io_context` of this session.
+     *
+     * Defers sending JSON-RPC event until it's subscription id is sent.
+     */
+    virtual void post(std::function<void()> cb) = 0;
```

Comment:
I think there is some problem with abstraction level here because Session interface isn't supposed to know of neither boost::asio nor json rpc. Because of this it's quite unclear here how post() is related to deferrring sending of a JSON RPC event until subscribtion ID is sent.

Reply:
- #1474